Framework/Beamline/src/ComponentInfo.cpp

```diff
@@ -111,16 +111,45 @@ Eigen::Quaterniond ComponentInfo::rotation(const size_t componentIndex) const {
   return (*m_rotations)[rangesIndex];
 }
 
+/**
+ * Extract the position of a component relative to it's parent
+ *
+ * The parent rotatation is unwound prior to establishing the offset. This means
+ *that
+ * recorded relative positions are independent of changes in rotation.
+ *
+ * BEWARE that this method does not account for scaling factors as found in
+ *RectangularDetectors
+ * see Instrument::makeLegacyParmeterMap for correct handling of those cases.
+ *
+ * BEWARE of peformance on repeated calls to this method as the transformation
+ *has to
+ * be established every time, independent of whether the parent remains the
+ *same.
+ *
+ * @param componentIndex
+ * @return
+ */
```

Comment:
With so many pitfalls, maybe we want to remove this method (now or later), and definitely not provide Python exports?